app/Services/ContextService.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace App\Services;
+
+use App\Group;
+use Auth;
+use Route;
+use Illuminate\Support\Facades\Gate;
+
+/**
+ * This service is responsible to return the correct context the user is currently in.
+ * It can be either 
+ * - a group, 
+ * - a custom list of groups,
+ * - all groups a user is member of,
+ * - all public groups, 
+ * - all groups, even closed ones (admin feature).
+ */
+class ContextService
+{
+    /**
+     * Returns current context as a string, can be : 
+     * - all : admin overview of all discussions for example
+     * - my : all discussions in my groups
+     * - group : a specific group is shown to the user
+     * - user : a user profile is shown
+     * - admin : server admin area
+     */
+    public function get()
+    {
+        if (Route::is('users.*')) {
+            return 'user';
+        }
+
+        if (Route::is('admin.*')) {
+            return 'admin';
+        }
+
+
+        $group = Route::getCurrentRoute()->parameter('group');
+
+        if ($group && $group->exists) {
+            return 'group';
+        }
+        // If not we need to show some kind of overview
+        else {
+            if (Auth::check()) {
+                if (Auth::user()->getPreference('show', 'my') == 'all' && Auth::user()->isAdmin()) {
+                    return 'all';
+                }
+                if (Auth::user()->getPreference('show', 'my') == 'public') {
+                    return 'public';
+                }
+                if (Auth::user()->getPreference('show', 'my') == 'my') {
+                    return 'my';
+                }
+            }
+        }
+        return 'public';
+    }
+
+    /**
+     * Return true if the passed $context is the current content.
+     * $context can be a Group model or 'my', 'public', 'all', 'group', 'user'
+     */
+    public function is($context)
+    {
+        if ($context instanceof Group) {
+            $group = Route::getCurrentRoute()->parameter('group');
+            return $group && $context->id == $group->id;
```

Comment:
It is generally desirable to avoid multiple return statements. Unless you are doing so to make a significant performance gain, it's generally a good way to introduce bugs, because later developers easily lose track of them.

Reply:
In this specific situation I don't find a clean way, but I'll keep it in mind.